internal/store/elasticsearch/discovery_search_repository_test.go

```diff
@@ -275,3 +275,187 @@ func loadTestFixture(cli *elasticsearch.Client, esClient *store.Client, filePath
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "table"},
+							{Name: "name", Value: "tablename-common"},
+						},
+						Assets: []asset.Asset{{Name: "tablename-common"}},
+					},
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "table"},
+							{Name: "name", Value: "tablename-mid"},
+						},
+						Assets: []asset.Asset{{Name: "tablename-mid"}},
+					},
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "topic"},
+							{Name: "name", Value: "consumer-mq-2"},
+						},
+						Assets: []asset.Asset{{Name: "consumer-mq-2"}},
+					},
+					{
+						Fields: []asset.GroupField{
+							{Name: "name", Value: "consumer-topic"},
+							{Name: "type", Value: "topic"},
+						},
+						Assets: []asset.Asset{{Name: "consumer-topic"}},
+					},
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "topic"},
+							{Name: "name", Value: "order-topic"},
+						},
+						Assets: []asset.Asset{{Name: "order-topic"}},
+					},
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "topic"},
+							{Name: "name", Value: "purchase-topic"},
+						},
+						Assets: []asset.Asset{{Name: "purchase-topic"}},
+					},
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "topic"},
+							{Name: "name", Value: "transaction"},
+						},
+						Assets: []asset.Asset{{Name: "transaction"}}},
+				},
+			},
+			{
+				Description: "should group assets which match group by fields",
+
+				Config: asset.GroupConfig{
+					GroupBy:        []string{"type"},
+					IncludedFields: []string{"name"},
+				},
+				Expected: []asset.GroupResult{
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "table"},
+						},
+						Assets: []asset.Asset{
+							{Name: "tablename-1"},
+						},
+					},
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "topic"},
+						},
+						Assets: []asset.Asset{
+							{Name: "order-topic"},
+						},
+					},
+				},
+			},
+			{
+				Description: "should not return assets without fields specified in filters",
+				Config: asset.GroupConfig{
+					GroupBy: []string{"type"},
+					Filters: map[string][]string{
+						"data.country":     {"id"},
+						"data.environment": {"production"},
+						"data.company":     {"gotocompany"},
+					},
+					IncludedFields: []string{"name"},
+				},
+				Expected: []asset.GroupResult{
+					{
+						Fields: []asset.GroupField{
+							{Name: "type", Value: "topic"}},
+						Assets: []asset.Asset{
+							{Name: "consumer-topic"},
+						},
+					},
+				},
+			},
+		}
+		for _, test := range tests {
+			t.Run(test.Description, func(t *testing.T) {
+				results, err := repo.GroupAssets(ctx, test.Config)
+				assert.NoError(t, err)
+				assert.Equal(t, len(test.Expected), len(results))
+
+				for i, res := range test.Expected {
+					assert.Equal(t, len(res.Fields), len(results[i].Fields))
```

Comment:
Why only assert on the length? Is the order different sometimes? If so, could we sort the fields (in the test) before comparison?

Reply:
Yes, the order is changing. Let me check on sorting

Reply:
done